### DISCUSSION TITLE
Error 400 Github AI Marketplace

### DISCUSSION THREAD

**Discussion by avcode-exe** [open-webui/open-webui#5461] [General]
# Bug Report

## Installation Method

`pip install open-webui`

## Environment

- **Open WebUI Version:** 0.3.21
- **Ollama (if applicable):** N/A

- **Operating System:** Windows 11 24H2 - WSL2 Ubuntu Preview
- **Browser (if applicable):** Arc 1.19.2 (49737) - Chromium Engine Version 128.0.6613.138

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Get a normal respond.

## Actual Behavior:

Return Error 400.

## Description

**Bug Summary:**
I received Error 400 when sending a message to a model from github marketplace, endpoint @ `https://models.inference.ai.azure.com`.

## Reproduction Details

**Steps to Reproduce:**
1. `pip install open-webui==0.3.21`
2. `open-webui serve`
3. Go to Admin Panel - Settings - Connections.
4. Add `https://models.inference.ai.azure.com` as the API Endpoint under OpenAI API and Github token as the API Key.
5. Go back to "New Chat" and select "gpt-4o-mini".
6. Send a message.

## Logs and Screenshots

**Browser Console Logs:**
![image](https://github.com/user-attachments/assets/b537ae6d-9448-4409-a459-e573736b058b)

**Docker Container Logs:**
N/A

**Screenshots/Screen Recordings (if applicable):**
![image](https://github.com/user-attachments/assets/4369390c-b3c7-44f4-9aa7-09d626432832)

## Additional Information

**Error Log:**
```
ERROR [open_webui.apps.openai.main] 400, message='Bad Request', url='https://models.inference.ai.azure.com/chat/completions'
Traceback (most recent call last):
  File "/home/alphabeta/miniconda3/envs/Open-WebUI-py311/lib/python3.11/site-packages/open_webui/apps/openai/main.py", line 414, in generate_chat_completion
    r.raise_for_status()
  File "/home/alphabeta/miniconda3/envs/Open-WebUI-py311/lib/python3.11/site-packages/aiohttp/client_reqrep.py", line 1093, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 400, message='Bad Request', url='https://models.inference.ai.azure.com/chat/completions'
INFO:     127.0.0.1:55136 - "POST /api/chat/completions HTTP/1.1" 400 Bad Request
INFO:     127.0.0.1:55136 - "POST /api/v1/chats/85371f10-1abb-4a9e-b8ab-a5f876c4183c HTTP/1.1" 200 OK
INFO:     127.0.0.1:55136 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 OK
generate_title
azureml://registries/azure-openai/models/gpt-4o-mini/versions/1
ERROR [open_webui.apps.openai.main] 400, message='Bad Request', url='https://models.inference.ai.azure.com/chat/completions'
Traceback (most recent call last):
  File "/home/alphabeta/miniconda3/envs/Open-WebUI-py311/lib/python3.11/site-packages/open_webui/apps/openai/main.py", line 414, in generate_chat_completion
    r.raise_for_status()
  File "/home/alphabeta/miniconda3/envs/Open-WebUI-py311/lib/python3.11/site-packages/aiohttp/client_reqrep.py", line 1093, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 400, message='Bad Request', url='https://models.inference.ai.azure.com/chat/completions'
INFO:     127.0.0.1:55136 - "POST /api/task/title/completions HTTP/1.1" 400 Bad Request
```

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by achanyu1** [comment]
- Another repository on GitHub, [LiteLLM](https://github.com/BerriAI/litellm), supports all models on GitHub's marketplace. This is how to set up the proxy. [https://docs.litellm.ai/docs/providers/github](https://docs.litellm.ai/docs/providers/github)

- I had the same problem as you, apparently the call method provided by GitHub is not what he declares it to be.Here's the example he provides in codespace.


```bash
#!/bin/bash
curl -X POST "https://models.inference.ai.azure.com/chat/completions" \
    -H "Content-Type: application/json" \
    -H "Authorization: Bearer $GITHUB_TOKEN" \
    -d '{
        "messages": [
            {
                "role": "system",
                "content": "You are a helpful assistant."
            },
            {
                "role": "user",
                "content": "What is the capital of France?"
            }
        ],
        "model": "gpt-4o-mini"
    }'
```
- Have fun!
